ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -700,20 +718,47 @@ private void uploadClusterAdminInfos(Map<String, Map<String, Map<String, String>
    * @param adminConfigZNodePath the znode path of admin config.
    */
   private void deleteClusterAdminInfos(String clusterAdminType, String adminConfigZNodePath) {
-    Properties storeProps = new Properties();
-    storeProps.setProperty("helix.property.store.root.path", "/" + clusterName + "/" + PROPERTYSTORE_STR);
-    HelixPropertyStoreConfig propertyStoreConfig = new HelixPropertyStoreConfig(new VerifiableProperties(storeProps));
     for (Map.Entry<String, ClusterMapUtils.DcZkInfo> entry : dataCenterToZkAddress.entrySet()) {
       info("Deleting {} infos for datacenter {}.", clusterAdminType, entry.getKey());
-      HelixPropertyStore<ZNRecord> helixPropertyStore =
-          CommonUtils.createHelixPropertyStore(entry.getValue().getZkConnectStrs().get(0), propertyStoreConfig, null);
+      HelixPropertyStore<ZNRecord> helixPropertyStore = createHelixPropertyStore(entry.getKey());
       if (!helixPropertyStore.remove(adminConfigZNodePath, AccessOption.PERSISTENT)) {
         logger.error("Failed to remove {} infos from datacenter {}", clusterAdminType, entry.getKey());
       }
       helixPropertyStore.stop();
     }
   }
 
+  /**
+   * Convert instance configs to the new DataNodeConfig format and persist them in the property store.
+   */
+  private void migrateToPropertyStore() {
+    adminForDc.forEach((dcName, helixAdmin) -> {
```

Comment:
Optional: considering making this multi-threaded (one thread  per colo). 

(Never mind, it should be fine, as it's one-off operation)

Reply:
Parallelism is a good idea anyways. I made this multi-threaded since it doesn't add much complexity to the code.